source/core/src/main/com/csse3200/game/components/tasks/ChaseTask.java

```diff
@@ -12,14 +12,14 @@
 
 /** Chases a target entity until they get too far away or line of sight is lost */
 public class ChaseTask extends DefaultTask implements PriorityTask {
-  private final Entity target;
-  private final int priority;
-  private final float viewDistance;
-  private final float maxChaseDistance;
-  private final PhysicsEngine physics;
-  private final DebugRenderer debugRenderer;
-  private final RaycastHit hit = new RaycastHit();
-  private MovementTask movementTask;
+  protected final Entity target;
```

Comment:
Pause, Avoid and Chase have a large overlap and would require a lot of duplicated code as flagged per sonar lint. So I made it that pause and avoid inheritted from chase (base code) meaning the things I was using needed to be protected rather than private.